README.md

```diff
@@ -1,4 +1,184 @@
+As soon as you open the dashboard, it will search for a file `config.json`
+containing the following keys
+
+* __astarte_api_url__ (required)
+  The base URL of your Astarte API endpoints.
+  This will be used to deduct the endpoints of all Astarte components
+
+  + AppEngine: api_url + /appengine
+  + Realm Management: api_url + /realmmanamenet
+  + Pairing: api_url + /pairing
+  + Flow: api_url + /flow
+
+  If the special string `localhost` is set, it will set the Astarte in 5 minutes
+  default local configuration
+
+  + AppEngine: `http://localhost:4002`
+  + Realm Management: `http://localhost:4000`
+  + Pairing: `http://localhost:4003`
+  + Flow: `http://localhost:4009`
+
+  In custom deployments those URLs may change, so you can overwrite any of those
+  with the following optionals keys
+  + __realm_management_url__
+  + __appengine_url__
+  + __pairing_url__
+  + __flow_url__
+
+* __default_realm__ (optional)
+  The default realm to login into
+
+* __enable_flow_preview__ (optional)
+  This requires your Astarte cluster to have Flow configured and running.
+  When enabled (set it to `true`) the dashboard will display flow API status
+  and relative pages like flows, pipelines and blocks.
+
+* __auth__ (required)
+  The list of auth options to be accepted for login.
+
+  Supported authentications methods are: Direct token or OAuth2 implicit grant
+
+  + __oauth__
+    Following the OAuth standard, the Dashboard will redirect you to your OAuth provider
+    where you'll be able to login normally. On succesfull login you'll be redirected back
+    to the dashboard.
+
+  + __token__
+    The user will be asked for a token to be used with the Astarte API.
+    You can generate one from the realm private key using [astartectl](https://github.com/astarte-platform/astartectl)
+
+  If multiple auth options are enabled, the user will have a choice on which one to use.
+  Example configuration:
+
+* __default_auth__ (required)
+  The default auth option to display when attempting login.
+
+
+An example config would look like:
+```json
+{
+  "astarte_api_url": "https://api.example.com",
+  "enable_flow_preview": true,
+  "default_realm": "myrealm",
+  "default_auth": "token",
+  "auth": [
+    {
+      "type": "token"
+    },
+    {
+      "type": "oauth",
+      "oauth_api_url": "https://auth.example.com"
+    }
+  ]
+}
+```
+
+### Starting up a local server
+
+1. Clone this repo locally
+   ```sh
+   git clone git@github.com:astarte-platform/astarte-dashboard.git && cd astarte-dashboard
+   ```
+2. Install the project dependencies
+   ```sh
+   npm install
+   ```
+3. Place your configuration file in `src/user-config/config.json`
+4. Start the dev server
+   ```sh
+   npm run start
+   ```
+5. Open the browser at the displayed URL, usually `http://localhost:8080`
+
+
+### Testing
+
+To run tests in the CLI you can simply run the command `test`
+```sh
+npm run test
```

Comment:
`npm run test` will probably fail, as will tests run with `npm run cypress:open`.
You should specify that `npm run start-ci` should be run first.

Reply:
good point!